packages/design/src/DataTable/Cells.tsx

```diff
@@ -80,14 +81,39 @@ export const DateCell = ({ data }: { data: Date }) => (
   <Cell>{displayDate(data)}</Cell>
 );
 
+const TextEllipsisLabelContent = ({
+  text,
+  maxLength,
+}: {
+  text: string;
+  maxLength: number;
+}) => {
+  let displayText = text;
+  const isLonger = text.length > maxLength;
+
+  if (isLonger) {
+    displayText = text.substring(0, 100);
+  }
+
+  return (
+    <LabelContent title={text} dots={isLonger}>
+      {displayText}
+    </LabelContent>
+  );
+};
+
 const renderLabelCell = (labels: string[] = []) => {
   const $labels = labels.map(label => (
-    <Label mb="1" mr="1" key={label} kind="secondary">
-      {label}
+    <Label mr="1" key={label} kind="secondary">
+      <TextEllipsisLabelContent text={label} maxLength={100} />
     </Label>
   ));
 
-  return <Cell>{$labels}</Cell>;
+  return (
+    <Cell>
+      <div css={{ lineHeight: '20px' }}>{$labels}</div>
```

Comment:
since we opted for margin
```suggestion
      <div>{$labels}</div>
```